x-pack/plugins/apm/public/utils/formatters.ts

```diff
@@ -141,3 +142,32 @@ export function asPercent(
   const decimal = numerator / denominator;
   return numeral(decimal).format('0.0%');
 }
+
+export function roundToPlaces(value: number, places: number) {
+  const multiplier = 10 * places;
+  return Math.floor(value * multiplier) / multiplier;
+}
+
+export function asBytes(value: number | null, places = 2) {
+  if (value === null || isNaN(value)) {
+    return '';
+  }
+
+  if (value > 1e12) {
+    return `${roundToPlaces(value / 1e12, places)} TB`;
+  }
+
+  if (value > 1e9) {
+    return `${roundToPlaces(value / 1e9, places)} GB`;
+  }
+
+  if (value > 1e6) {
+    return `${roundToPlaces(value / 1e6, places)} MB`;
+  }
+
+  if (value > 1000) {
+    return `${roundToPlaces(value / 1000, places)} KB`;
+  }
+
+  return `${roundToPlaces(value, places)} B`;
+}
```

Comment:
Oh, sorry for coming with this so late - I just realize this will end up with different tick units. Can we solve it like we did with `getTimeFormatter`:
https://github.com/elastic/kibana/blob/7c386dd120fba004caefd5c196137910b7633d83/x-pack/plugins/apm/public/utils/formatters.ts#L81-L91

Something like:
```js
const byteFormatter = getByteFormatter(max);
```